### PR TITLE
Issue 462

### PR DESCRIPTION
**Pull request by panchew**

### CODE REVIEW

**Comment by mvl22** [comment]
@nikolai-b Can you review.

File: app/controllers/message_thread/subscriptions_controller.rb [hunk 1]
```diff
@@ -2,18 +2,26 @@ class MessageThread::SubscriptionsController < MessageThread::BaseController
...
+        set_flash_message :success
+        format.html { redirect_to thread_path @thread }
+        format.js   { }
+      else
+        set_flash_message :failure
+        format.html { redirect_to thread_path @thread }
+        format.js   { }
+      end
     end
-    redirect_to thread_path @thread
```
Comment (nikolai-b) [inline app/controllers/message_thread/subscriptions_controller.rb]: Can't you just have the format after the `if`
```ruby
if  @thread.add_subscriber current_user
  set_flash_message :success
else
  set_flash_message :failure
end
format.html { redirect_to thread_path @thread }
format.js   { }
```
?

**Comment by nikolai-b** [comment]
Same deal with the schema https://github.com/cyclestreets/cyclescape/pull/475#issuecomment-170088056 but :+1: looks good!

File: app/controllers/message_thread/subscriptions_controller.rb (hunk 1 above)
Comment (panchew) [inline app/controllers/message_thread/subscriptions_controller.rb]: Makes sense; I've pushed that suggestion. Thanks

**Comment by panchew** [comment]
I've removed changes to schema.rb and DRY-ed the respond_to code in subscriptions_controller

**Comment by mvl22** [comment]
Sorry, where does this appear? I was expecting buttons in the listing at ```/threads```.

**Pull request merged**